### PR TITLE
Fix "All domains list" header style (WIP)

### PR DESCRIPTION
**Pull request by gius80**
## Changes proposed in this Pull Request

This PR addresses some style fixes for the new header in All domains list:

* Height should be 70px, currently it’s 72px with some extra margin on top which makes it look not centered.
* It should be “sticky” when scrolling.
* The color of “?” next to the title should be var(–studio-gray-30).
* Filter dropdown: In single site view “All my domains” isn’t centered.
* Mobile: height should be 60px on mobile (*Update from Nov. 29*)
* Mobile: “3 dots” menu should also be 40 x 40px, with no border.
* Mobile: margin between the header and the filter dropdown should be 16px, currently it’s bigger.

pcYYhz-pX-p2#comment-356
pcYYhz-pX-p2#comment-354

## Screenshot

header-desktop

header-mobile

## Testing instructions

* Build this branch locally or open the live Calypso link
* Go to "Upgrades > Domains" _(header it's same for All sites/Single site)_
* If you're in the live Calypso link, please append ?flags=domains/management-list-redesign to your URL to enable the appropriate feature flag
* Verify that the style of the header meets all the feedbacks above (use screenshots as reference)

### CODE REVIEW

File: client/components/select-dropdown/style.scss [hunk 1]
```diff
@@ -279,7 +279,7 @@ $compact-header-height: 35;
 .select-dropdown__separator {
 	border-top: 1px solid var( --color-neutral-10 );
 	display: block;
-	margin: 8px 0;
+	margin: 8px 0 0;
```
Comment (leonardost) [inline client/components/select-dropdown/style.scss]: Generally we should avoid directly changing the base style of a component that's widely used because it might break the layout somewhere else we don't expect 🙂

What we usually do is overwrite that specific CSS rule inside the component we're working on, so that rule will only apply for our component. So instead of editing `select-dropdown/style.scss`, we could add this rule to `breadcrumbs/style.scss`:
```
.breadcrumbs {
  ...
  .select-dropdown__separator {
    margin: 8px 0 0;
  }
}
```

File: client/my-sites/domains/domain-management/components/breadcrumbs/index.jsx [hunk 2]
```diff
@@ -128,13 +128,15 @@ const Breadcrumbs = ( { items, mobileItem, buttons, mobileButtons, className } )
 	);
 
 	return (
-		<div className={ classNames( 'breadcrumbs', className ) }>
-			<div className="breadcrumbs__content">
-				{ renderItems() }
-				{ renderButtons() }
+		<React.Fragment>
+			<div className={ classNames( 'breadcrumbs', className ) }>
```
Comment (leonardost) [inline client/my-sites/domains/domain-management/components/breadcrumbs/index.jsx]: In a similar line of thought as the first comment, the root element of a component should be the one that contains the root CSS class `breadcrumbs`. That way, the CSS classes will keep their cascading logic (`.breadcrumbs > .breadcrumbs__content`, etc) and if other people use this component, they can more easily modify its style as needed.

So it would become something like:
```
<div className={ classNames( 'breadcrumbs', className ) }>
	<div>
		<div className="breadcrumbs__content">
			{ renderItems() }
			{ renderButtons() }
		</div>
	</div>
	<div className="breadcrumbs__bottom-border"></div>
</div>
```

We can probably remove that extra `div` around `.breadcrumbs__content` 🤔 

In other words, what I mean is that the `.breadcrumbs__spacer` div should be inside `.breadcrumbs` 😆

File: client/my-sites/domains/domain-management/components/breadcrumbs/style.scss [hunk 3]
```diff
@@ -2,59 +2,40 @@
...
-	position: relative;
+	position: fixed;
 	margin: 0;
 	z-index: 10;
+	background-color: white;
+	top: var( --masterbar-height );
+	padding: 0 24px;
+	left: 0;
+	right: 0px;
+	border-bottom: 1px solid #dcdcde;
```
Comment (leonardost) [inline client/my-sites/domains/domain-management/components/breadcrumbs/style.scss]: I know it was like this before (I was the one who did this 😆), but we should try using variables for color where possible - in this case, `#dcdcde` is equivalent to `var( --studio-gray-5 )`.

The other day I found this nice codepen with WordPress.org's color palette: https://codepen.io/ryelle/full/WNGVEjw

**Review by leonardost — CHANGES_REQUESTED** [review]
Thanks for these fixes and great job on the sticky breadcrumb bar! It looks really nice! 😄 

There are just some observations that I think should be taken care of before we can merge this 🙂 besides the comments I left in the code, I noticed two other things:

Markup on 2021-11-30 at 15:54:55

It seems there's still a little more space above the component than below - maybe the spacer should have a height of around 78px for mobile?

Screen Shot 2021-11-30 at 15 28 29

Another point is It seems the left and right paddings in mobile view were changed - was this intended? They're 24px here, but were 16px before:

Screen Shot 2021-11-30 at 15 29 03

**Comment by gius80** [comment]
Hi @poligilad-auto, I need a couple of clarification here.

> Mobile: margin between the header and the filter dropdown should be 16px, currently it’s bigger.

I modified only the bottom margin, but as @leonardost rightly observed, now the upper margin it's bigger. Should I equalize both margins to 16px?

Looking at the layouts on Figma I noticed that the side padding of the header should be 16px instead of 24px. Should I leave it 24px or fix it?

Thanks!

**Comment by poligilad-auto** [comment]
> I modified only the bottom margin, but as @leonardost rightly observed, now the upper margin it's bigger. Should I equalize both margins to 16px?

Yes, exactly.

> Looking at the layouts on Figma I noticed that the side padding of the header should be 16px instead of 24px. Should I leave it 24px or fix it?

You're right. It should be 16px in mobile as that's the padding for everything else.

**Another small thing and I apologize as I [changed my comment later on](https://wp.me/pcYYhz-pX#comment-354)**: Mobile: height of the header should be 60px on mobile.

File: client/my-sites/domains/domain-management/components/breadcrumbs/style.scss (hunk 3 above)
Comment (gius80) [reply client/my-sites/domains/domain-management/components/breadcrumbs/style.scss]: Cool, very useful!

File: client/my-sites/domains/domain-management/components/breadcrumbs/index.jsx (hunk 2 above)
Comment (gius80) [reply client/my-sites/domains/domain-management/components/breadcrumbs/index.jsx]: You're totally right, I needed outside because `.breadcrumbs` is fixed and the spacer should not. I made a small refactoring. Good catch!

File: client/components/select-dropdown/style.scss (hunk 1 above)
Comment (gius80) [reply client/components/select-dropdown/style.scss]: I agree 😅

File: client/my-sites/domains/domain-management/components/breadcrumbs/index.jsx [hunk 4]
```diff
@@ -128,13 +128,17 @@ const Breadcrumbs = ( { items, mobileItem, buttons, mobileButtons, className } )
...
+					<div className="breadcrumbs__content">
+						{ renderItems() }
+						{ renderButtons() }
+					</div>
+				</div>
+				<div className="breadcrumbs__spacer"></div>
 			</div>
-			<div className="breadcrumbs__bottom-border"></div>
-		</div>
+		</React.Fragment>
```
Comment (leonardost) [inline client/my-sites/domains/domain-management/components/breadcrumbs/index.jsx:141]: With the refactoring, this `React.Fragment` can be removed 🙂

File: client/my-sites/domains/domain-management/components/breadcrumbs/style.scss [hunk 5]
```diff
@@ -190,3 +188,15 @@
...
+
+.breadcrumbs__spacer {
+	height: 58px;
+
+	@include breakpoint-deprecated( '<960px' ) {
+		height: 67px;
+	}
+
+	@include breakpoint-deprecated( '<660px' ) {
+		height: calc( 76px );
```
Comment (leonardost) [inline client/my-sites/domains/domain-management/components/breadcrumbs/style.scss:200]: Is this `calc` needed here?

File: client/my-sites/domains/domain-management/components/breadcrumbs/style.scss [hunk 6]
```diff
@@ -190,3 +188,15 @@
 		display: flex;
 	}
 }
+
+.breadcrumbs__spacer {
+	height: 58px;
+
+	@include breakpoint-deprecated( '<960px' ) {
+		height: 67px;
+	}
```
Comment (leonardost) [inline client/my-sites/domains/domain-management/components/breadcrumbs/style.scss]: I was wondering about this intermediary breakpoint - do we need it? Or do we need only one height for mobile and one for desktop views?

File: client/my-sites/domains/domain-management/components/breadcrumbs/style.scss (hunk 5 above)
Comment (leonardost) [inline client/my-sites/domains/domain-management/components/breadcrumbs/style.scss]: Is this calc needed here?

File: client/my-sites/domains/domain-management/components/breadcrumbs/style.scss (hunk 6 above)
Comment (leonardost) [reply client/my-sites/domains/domain-management/components/breadcrumbs/style.scss]: If we only need these two, we could use the `@include break-mobile` media query instead of the `breakpoint-deprecated` one, something like this:
```
.breadcrumbs__spacer {
  height: 76px;

  @include break-mobile {
    height: 58px;
  }
}
```

File: client/my-sites/domains/domain-management/components/breadcrumbs/style.scss [hunk 7]
```diff
@@ -2,59 +2,54 @@
...
 
-	@include break-medium {
-		position: sticky;
-		top: -44px;
-		margin-top: -44px;
+	.select-dropdown__separator {
+		border-top: 1px solid var( --color-neutral-10 );
+		display: block;
+		margin: 8px 0 0;
 	}
```
Comment (leonardost) [inline client/my-sites/domains/domain-management/components/breadcrumbs/style.scss:37]: Nice! 👍

File: client/my-sites/domains/domain-management/components/breadcrumbs/index.jsx (hunk 4 above)
Comment (leonardost) [inline client/my-sites/domains/domain-management/components/breadcrumbs/index.jsx]: With the refactoring, this outer React.Fragment can be removed 🙂

**Review by leonardost — APPROVED** [review]
LGTM @gius80, nice job! Thanks for the patience and for updating the aforementioned points 😄 left some minor comments that you can address before merging this, but otherwise it looks good! 👍 

Edit: There's also a code conflict that needs to be solved before merging 🙂

File: client/my-sites/domains/domain-management/components/breadcrumbs/style.scss (hunk 5 above)
Comment (gius80) [reply client/my-sites/domains/domain-management/components/breadcrumbs/style.scss]: Ops.. 😅

File: client/my-sites/domains/domain-management/components/breadcrumbs/style.scss (hunk 6 above)
Comment (gius80) [reply client/my-sites/domains/domain-management/components/breadcrumbs/style.scss]: Yes we need this because the #content wrapper change padding value at 960px, from 71px to 79px, and the spacer needs to compensate this difference (there are some weirds changes in the UI depending on old mq 😅). BTW I change from 67px to 66px.

**Comment by poligilad-auto** [comment]
Perfect @gius80! Sticky header working well and also 60px height on mobile looks great 👌

File: client/my-sites/domains/domain-management/components/breadcrumbs/style.scss (hunk 6 above)
Comment (leonardost) [reply client/my-sites/domains/domain-management/components/breadcrumbs/style.scss]: Got it, good call!
